extensions/http-curl/client/HTTPClient.cpp

```diff
@@ -328,18 +325,15 @@ void HTTPClient::setRequestHeader(std::string key, std::optional<std::string> va
 }
 
 namespace {
-struct CurlSListFreeAll {
-  void operator()(struct curl_slist* slist) const {
-    curl_slist_free_all(slist);
-  }
-};
+using CurlSlistDeleter = decltype([](struct curl_slist* slist) { curl_slist_free_all(slist); });
```

Comment:
if the only purpose of `CurlSlistDeleter` is to be used in a `std::unique_ptr` it might be better to create a `CurlSlistUniquePtr` typedef

Reply:
I don't really think saving a few characters of typing with the alias is worth the extra mental indirection, I prefer spelling out the actual type of `std::unique_ptr<struct curl_slist, CurlSlistDeleter>` in this case.